internal/host/plugin/repository_host_set_test.go

```diff
@@ -1321,6 +1321,10 @@ func TestRepository_Endpoints(t *testing.T) {
 	TestSetMembers(t, conn, hostSet192.GetPublicId(), []*Host{h1})
 	TestSetMembers(t, conn, hostSet100.GetPublicId(), []*Host{h1})
 
+	hostSetDNS := TestSet(t, conn, kms, sched, catalog, plgm, WithName("hostSetDNS"), WithPreferredEndpoints([]string{"cidr:5.4.0.0/16"}))
+	h2 := TestHost(t, conn, catalog.GetPublicId(), "dns", withIpAddresses([]string{"5.4.3.2"}), withDnsNames([]string{"example.com"}))
+	TestSetMembers(t, conn, hostSetDNS.GetPublicId(), []*Host{h2})
```

Comment:
Thank you so much for identifying this gap in our testing.  I'd like to ensure we also verify the selection of the DNS address that might be on a host.
To make this new test case be closer to the other tests in this same file, instead of creating a new host here, could you add the `withDnsNames([]string{"example.com"})` to the host created on line 1319 and then move your hostset alongside the ones already created in lines 1314-1316 and have it uses `WithPreferredEndpoints([]string{"dns:*"})`?  That'll not only add coverage to the line that you fixed in this PR, but it'll also ensure that the DNS name on the host is properly selected and returned.